nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/util/AbstractStoreScanner.java

```diff
@@ -0,0 +1,175 @@
+        }
+
+        File parentFile = file.getParentFile();
+        if (!parentFile.exists() || !parentFile.isDirectory()) {
+            throw new IllegalArgumentException(String.format("error obtaining %s dir", getFileType()));
+        }
+
+        _scanner = new Scanner();
+        _scanner.setScanDirs(Collections.singletonList(parentFile));
+        _scanner.setScanInterval(1);
+        _scanner.setReportDirs(false);
+        _scanner.setReportExistingFilesOnStartup(false);
+        _scanner.setScanDepth(1);
+        _scanner.addListener(this);
+        addBean(_scanner);
+    }
+
+    @Override
+    public void fileAdded(final String filename) {
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("added {}", filename);
+        }
+
+        if (this.file.toPath().toString().equals(filename)) {
+            this.reload();
+        }
+
+    }
+
+    @Override
+    public void fileChanged(final String filename) {
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("changed {}", filename);
+        }
+
+        if (this.file.toPath().toString().equals(filename)) {
+            this.reload();
+        }
+
+    }
+
+    @Override
+    public void fileRemoved(final String filename) {
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("removed {}", filename);
+        }
+
+        if (this.file.toPath().toString().equals(filename)) {
+            this.reload();
+        }
+
+    }
+
+    @ManagedOperation(
+            value = "Scan for changes in the SSL Keystore/Truststore",
+            impact = "ACTION"
+    )
+    public void scan() {
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("scanning");
+        }
+
+        this._scanner.scan();
+        this._scanner.scan();
+    }
+
+    @ManagedOperation(
+            value = "Reload the SSL Keystore/Truststore",
+            impact = "ACTION"
+    )
+    public void reload() {
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("reloading {} file {}", getFileType(), this.file);
+        }
+
+        try {
+            this.sslContextFactory.reload((scf) -> scf.setSslContext(createSSLContext(tlsConfiguration)));
+        } catch (final Throwable t) {
+            getLogger().warn("{} Reload Failed", StringUtils.capitalize(getFileType()), t);
+        }
+
+    }
+
+    @ManagedAttribute("scanning interval to detect changes which need reloaded")
+    public int getScanInterval() {
+        return this._scanner.getScanInterval();
+    }
+
+    public void setScanInterval(int scanInterval) {
+        this._scanner.setScanInterval(scanInterval);
+    }
+
+    protected SslContextFactory getSslContextFactory() {
+        return sslContextFactory;
+    }
+
+    protected abstract String getFileType();
+    protected abstract Resource getResource();
+    protected abstract Logger getLogger();
```

Comment:
Instead of making `getLogger()` abstract, the Logger can be defined as an instance variable with `getClass()` so that concrete classes will have distinct loggers.

Reply:
Got it. Will change it to use `getClass()`